lib/spreewald/email_steps.rb

```diff
@@ -69,9 +69,16 @@
   mail = @mail || ActionMailer::Base.deliveries.last
   index = { nil => 0, 'first' => 0, 'second' => 1, 'third' => 2 }[index_in_words]
   url_pattern = %r{(?:http|https)://[^/]+([^"'\s\\]*)}
-  mail_body = MailFinder.email_text_body(mail).to_s
-  only_path = mail_body.scan(url_pattern)[index][0]
-  visit only_path
+
+  paths = if mail.html_part
+    dom = Nokogiri::HTML(mail.html_part.body.to_s)
+    (dom / 'a[href]').map { |a| a['href'].match(url_pattern)[1] }
+  else
+    mail_body = MailFinder.email_text_body(mail).to_s
+    mail_body.scan(url_pattern)[0]
```

Comment:
It's a bit hard to parse/understand why you are using `.match(pattern)[1]` for HTML parts and `.scan(pattern)[0]` for text parts.

Reply:
Ok after reading the docs I get it, `match` includes the entire match as the array's first entry.
Please unify the use in this case :slightly_smiling_face:

Reply:
Valid point, will adjust :+1:

Reply:
The implementation is actually incorrect since I changed it to `scan(...)[0]`. Will fix.

Reply:
Actually can't use the same method in both cases since they handle different approaches (one loops over HTML hyperlinks, the other one parses the entire text body).

However, I've cleaned it up a bit and added a test for the text body part (and fixed it).